### PR TITLE
3.1 PI updates

### PR DESCRIPTION
**Pull request by magicbadger**

### CODE REVIEW

File: RadioDns-PlugIt/SPI/modules/aws_spi.py [hunk 1]
```diff
@@ -42,26 +42,48 @@ def on_pi_resource_changed(self, event_name, station):
...
+                            AWSSPI.get_static_bucket_pi_filename(
+                                1,
+                                channel.service_identifier,
+                                start_date.strftime("%Y%m%d")
+                            ),
+                            contents,
+                            station.id,
+                        )                         
+                    except AttributeError as e:
+                            print("[WARN][AWS] in pi resource changed handler", e)
```
Comment (ioannisNoukakis) [inline RadioDns-PlugIt/SPI/modules/aws_spi.py]: Bad indentation.

File: RadioDns-PlugIt/SPI/modules/aws_spi.py [hunk 2]
```diff
@@ -42,26 +42,48 @@ def on_pi_resource_changed(self, event_name, station):
...
+                contents = SPI.utils.generate_pi_file(start_date, "/templates/radioepg/schedule/xml3.html", station=station)
+                if not contents:
+                    continue
+
+                filtered = Channel.query.filter(Channel.station_id == station.id and Channel.type_id in ["fm", "dab"])
+                for channel in filtered:
+                    try:
+                        self.upload_file(
+                            AWSSPI.get_static_bucket_pi_filename(
+                                1,
```
Comment (ioannisNoukakis) [inline RadioDns-PlugIt/SPI/modules/aws_spi.py]: What about version 3=

File: RadioDns-PlugIt/SPI/utils.py [hunk 3]
```diff
@@ -306,6 +282,7 @@ def generate_pi_file(date, path=None, station=None):
 
     :return: The rendered XML file.
     """
+    print('generate_pi_file', path)
```
Comment (ioannisNoukakis) [inline RadioDns-PlugIt/SPI/utils.py]: Remove

File: RadioDns-PlugIt/SPI/utils.py [hunk 4]
```diff
@@ -288,7 +264,7 @@ def extract_pi_data_from_path(path, date):
     return None
 
 
-def generate_pi_file(date, path=None, station=None):
+def generate_pi_file(date, template_uri, path=None, station=None):
```
Comment (ioannisNoukakis) [inline RadioDns-PlugIt/SPI/utils.py]: template_uri is not described in the docstring.

File: RadioDns-PlugIt/SPI/modules/aws_spi.py (hunk 2 above)
Comment (magicbadger) [reply RadioDns-PlugIt/SPI/modules/aws_spi.py]: Its in there, both support for routes and generating.

File: RadioDns-PlugIt/SPI/modules/aws_spi.py (hunk 1 above)
Comment (magicbadger) [reply RadioDns-PlugIt/SPI/modules/aws_spi.py]: *smack*

File: RadioDns-PlugIt/SPI/utils.py (hunk 4 above)
Comment (magicbadger) [reply RadioDns-PlugIt/SPI/utils.py]: have added the docstring

File: RadioDns-PlugIt/SPI/utils.py (hunk 3 above)
Comment (magicbadger) [reply RadioDns-PlugIt/SPI/utils.py]: I have removed.

**Pull request merged**